### PR TITLE
Ameerul / FEQ-2595 Sell order missing banner when advertiser has no balance

### PR DESCRIPTION
**Pull request by ameerul-deriv**
- Added logic to check if user's balance is low to show low balance error message in buy sell form

https://github.com/user-attachments/assets/ec6cbd15-f44a-470f-bd2b-dbc0684a74b9

### CODE REVIEW

**Review by mayuran-deriv — APPROVED** [review]

File: src/components/BuySellForm/BuySellForm.tsx [hunk 1]
```diff
@@ -261,6 +263,17 @@ const BuySellForm = ({ advertId, isModalOpen, onRequestClose }: TBuySellFormProp
...
 
+    useEffect(() => {
+        if (
+            balanceData.balance !== undefined &&
+            minOrderAmountLimit !== undefined &&
+            isBuy &&
+            (balanceData.balance === 0 || balanceData.balance < minOrderAmountLimit)
+        )
+            setShowLowBalanceError(true);
+        else setShowLowBalanceError(false);
```
Comment (nada-deriv) [inline src/components/BuySellForm/BuySellForm.tsx:274]:
```suggestion
        const shouldShowError = (
            balanceData.balance !== undefined &&
            minOrderAmountLimit !== undefined &&
            isBuy &&
            (balanceData.balance === 0 || balanceData.balance < minOrderAmountLimit)
        );
        setShowLowBalanceError(shouldShowError);
```

will this work?

Comment (ameerul-deriv) [reply src/components/BuySellForm/BuySellForm.tsx:274]: yup sure, this can make it more readable

**Review by nada-deriv — APPROVED** [review]

File: src/components/BuySellForm/BuySellAmount/BuySellAmount.scss [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+            border-color: #eaeced;
+
+            & .deriv-input__field {
+                color: #999999;
+            }
+
+            &:hover {
+                border-color: #eaeced !important;
+            }
+        }
```
Comment (ameerul-deriv) [inline src/components/BuySellForm/BuySellAmount/BuySellAmount.scss:15]: i couldn't override the input on hover effect when we disable the field to match prod without !Important. I tried directly accessing the input that also didnt work

File: src/components/BuySellForm/BuySellAmount/BuySellAmount.scss [hunk 3]
```diff
@@ -1,4 +1,19 @@
 .buy-sell-amount {
```
Comment (ameerul-deriv) [inline src/components/BuySellForm/BuySellAmount/BuySellAmount.scss:1]: Now the fields and adding payment method card / selecting a payment method are disabled when you're balance is low. I also fixed the bug where if your available balance is low, you couldn't place an order from Buy Modal, but in prod, you can

Uploading Screen Recording 2024-08-14 at 11.12.13 AM.mov…

Screenshot 2024-08-14 at 11 19 17 AM

File: src/components/BuySellForm/BuySellAmount/BuySellAmount.scss (hunk 2 above)
Comment (farrah-deriv) [reply src/components/BuySellForm/BuySellAmount/BuySellAmount.scss:15]: can we pass `disabled` as variant?

Comment (ameerul-deriv) [reply src/components/BuySellForm/BuySellAmount/BuySellAmount.scss:15]: Screenshot 2024-08-14 at 1 21 49 PM

I've done that already, but it doesn't disable the hover styles 😓 nor does it change the colour of the input text

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**